NewsApp-main/NewsApp/Scenes/NewsDetail/ViewController/NewsDetailViewController.swift

```diff
@@ -0,0 +1,370 @@
+               showRetryPopup(title: "Opsiyonel Hata", message: "Bilgileri alırken bir sorun oluştu. Lütfen tekrar deneyin.")
+           } else {
+               viewModel.getNewsDetailList()
+               reloadData()
+           }
+       }
+
+    // MARK: Setup timer check new source list
+    private func setupAutoRefreshTimer() {
+        autoRefreshTimer = Timer.scheduledTimer(
+            timeInterval: 60.0,
+            target: self,
+            selector: #selector(autoRefreshTimerFired),
+            userInfo: nil,
+            repeats: true
+        )
+    }
+
+    // MARK: Check new source list
+    @objc private func autoRefreshTimerFired() {
+        let previousCount = newsListCount
+        viewModel.getNewsDetailList()
+        if previousCount != viewModel.sourceDetailList.count {
+            reloadData()
+        } else {
+            print("previous list count : \(previousCount) , new list count : \(viewModel.sourceDetailList.count)")
+        }
+    }
+
+    // MARK: autoScroll
+    @objc private func autoScroll() {
+        let nextPage = (pageController.currentPage + 1) % pageController.numberOfPages
+        pageController.currentPage = nextPage
+
+        let offsetX = CGFloat(nextPage) * sliderCollectionView.bounds.width
+        sliderCollectionView.setContentOffset(CGPoint(x: offsetX, y: 0), animated: true)
+    }
+
+    // MARK: Pull to refresh source list data
+    @objc private func refreshData() {
+        checkAndShowRetryPopup()
+    }
+
+    // MARK: saveSlideItems
+    @objc func saveSlideItems(_ sender: UIButton) {
+        let indexPath = IndexPath(row: sender.tag, section: 0)
+        let selectedNews = viewModel.topNews[indexPath.row]
+
+        viewModel.toggleReadingListStatus(for: selectedNews)
+        sliderCollectionView.reloadData()
+    }
+
+    // MARK: saveNewsItems
+    @objc func saveNewsItems(_ sender: UIButton) {
+        let indexPath = IndexPath(row: sender.tag, section: 0)
+        let selectedNews = viewModel.sourceDetailList[indexPath.row]
+
+        viewModel.toggleReadingListStatus(for: selectedNews)
+        newsListCollectionView.reloadData()
+    }
+}
+
+// MARK: UICollectionView Delegate
+extension NewsDetailViewController: UICollectionViewDelegate, UICollectionViewDataSource, UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return collectionView == sliderCollectionView ? viewModel.topNews.count : viewModel.sourceDetailList.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == sliderCollectionView {
+            guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: NewsDetailCollectionViewCell.identifier, for: indexPath) as? NewsDetailCollectionViewCell else {
+                return UICollectionViewCell()
```

Comment:
Bu register olmadığı için crash verir. Case'lerde çok sorun değil ama aklıma geldi. Biz dummy bir EmptyCell oluşturuyoruz böyle durumlarda en azından crash olmasın diye.